pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -53,16 +55,32 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
 
     _readersByIndex = new HashMap<>();
     for (IndexType<?, ?, ?> indexType : IndexService.getInstance().getAllIndexes()) {
-      if (segmentReader.hasIndexFor(columnName, indexType)) {
-        IndexReaderFactory<?> readerProvider = indexType.getReaderFactory();
-        try {
-          IndexReader reader = readerProvider.createIndexReader(segmentReader, fieldIndexConfigs, metadata);
-          if (reader != null) {
-            _readersByIndex.put(indexType, reader);
-          }
-        } catch (IndexReaderConstraintException ex) {
-          LOGGER.warn("Constraint violation when indexing " + columnName + " with " + indexType + " index", ex);
+      Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
+      if (!indexType.getId().equals(StandardIndexes.TEXT_ID)) {
+        prepareIndexReader(segmentReader, indexType, fieldIndexConfigs, metadata);
+      } else if (IndexLoadingConfig.processExistingSegments(columnName, columnProperties)) {
```

Comment:
This is same as previous commit only. If we want to keep this logic, the previous one was more compact:
`if (!indexType.getId().equals(StandardIndexes.TEXT_ID) || IndexLoadingConfig.processExistingSegments(columnName, columnProperties))`